src/main/java/com/cleanroommc/groovyscript/registry/VirtualizedRegistry.java

```diff
@@ -2,25 +2,22 @@
 
 import com.cleanroommc.groovyscript.api.GroovyBlacklist;
 import com.cleanroommc.groovyscript.api.IScriptReloadable;
-import com.cleanroommc.groovyscript.helper.Alias;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.Nullable;
 
-import java.util.*;
-import java.util.stream.Collectors;
+import java.util.Collection;
 
 public abstract class VirtualizedRegistry<R> extends NamedRegistry implements IScriptReloadable {
 
-    private Collection<R> backup, scripted;
+    private final AbstractReloadableStorage<R> recipeStorage;
```

Comment:
Im thinking it might be better to have `VirtualizedRegistry` extend `AbstractReloadableStorage`?

Reply:
the current code structure requires that `VirtualizedRegistry` extends `NamedRegistry`. since `NamedRegistry` definitely shouldnt extend `AbstractReloadableStorage`, this plan isnt viable.